crates/cext/src/transpiler/passes/unitary_synthesis.rs

```diff
@@ -0,0 +1,222 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashSet;
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::passes::run_unitary_synthesis;
+use qiskit_transpiler::target::Target;
+
+/// @ingroup QkTranspilerPasses
+/// Run the UnitarySynthesis transpiler pass
+///
+/// The UnitarySynthesis transpiler pass will synthesis any UnitaryGates in the circuit into gates
+/// available in the target. The unitary gate will not be synthesize
+///
+/// Right now from C this pass only supports 1 and 2 qubit UnitaryGates, larger unitary matrices
+/// will be supported in a future release.
+///
+/// @param circuit A pointer to the circuit to run run UnitarySynthesis on
+/// @param target A pointer to the target to run the UnitarySynthesis on
+/// @param min_qubits The minimum number of qubits in the unitary to synthesize. If this is set
```

Comment:
`min_qubits` is not optional so I would rephrase to something like: 
```text
If the unitary is less than the specified number of qubits it will not be synthesized.
```